Release notes for V3.0.0-preview2 (#1034)
dotnet/SqlClient: CHANGELOG.md

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)

## [Preview Release 3.0.0-preview2.21105.1] - 2021-04-15

### Breaking Changes over preview release v3.0.0-preview1

> - `User Id` connection property now accepts `Client Id` instead of `Object Id` for **User-Assigned Managed Identity** [#1010](https://github.com/dotnet/SqlClient/pull/1010)

Reviewer: accepts ??  does it work with both? Otherwise is should maybe be "uses" ???

> - `User Id` connection property now accepts `Client Id` instead of `Object Id` for **User-Assigned Managed Identity** [#1010](https://github.com/dotnet/SqlClient/pull/1010)

Reviewer 2: I'd say requires.
```suggestion
- `User Id` connection property now requires `Client Id` instead of `Object Id` for **User-Assigned Managed Identity** [#1010](https://github.com/dotnet/SqlClient/pull/1010)
```

- `SqlDataReader` now returns a `DBNull` value instead of an empty `byte[]`. Legacy behavior can be enabled by setting `AppContext` switch **Switch.Microsoft.Data.SqlClient.LegacyRowVersionNullBehavior** [#998](https://github.com/dotnet/SqlClient/pull/998)

### Added

> - **Microsoft.Data.SqlClient** now depends on **Azure.Identity** library to acquire token for "Active Directory Managed Identity/MSI" and "Active Directory Service Principal" authentication modes. [#1010](https://github.com/dotnet/SqlClient/pull/1010)

Reviewer 2: ```suggestion
- **Microsoft.Data.SqlClient** now depends on **Azure.Identity** library to acquire a token for "Active Directory Managed Identity/MSI" and "Active Directory Service Principal" authentication modes. [#1010](https://github.com/dotnet/SqlClient/pull/1010)
```

- Upgraded Native SNI dependency to **v3.0.0-preview1** along with enhanced event tracing support [#1006](https://github.com/dotnet/SqlClient/pull/1006)

### Fixed

> - Fixed wrong data blended with transactions in .NET Core by marking a connection with open result to be doomed [#1023](https://github.com/dotnet/SqlClient/pull/1023)

Reviewer 2: ?
```suggestion
- Fixed wrong data blended with transactions in .NET Core by marking a connection as doomed if the transaction completes or aborts while there is an open result set[#1023](https://github.com/dotnet/SqlClient/pull/1023)
```

> - Fixed derived parameters containing typename incorrectly [#1020](https://github.com/dotnet/SqlClient/pull/1020)

Reviewer 2: ```suggestion
- Fixed derived parameters containing incorrect typename [#1020](https://github.com/dotnet/SqlClient/pull/1020)
```

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)

## [Preview Release 3.0.0-preview2.21105.1] - 2021-04-15

> ### Breaking Changes over Preview V3.0.0-preview1

Reviewer 3: should not this be preview2?
